service.toml

```diff
@@ -2,11 +2,41 @@ name = "minio"
 
 [namespace.storage]
 
+[namespace.storage.new]
+required = ["endpoint", "name", "credential"]
+optional = ["work_dir", "use_ssl", "region"]
+
 [namespace.storage.op.list]
-optional = ["list_mode"]
+optional = ["list_mode", "continuation_token", "list_buffer_size"]
 
 [namespace.storage.op.read]
-optional = ["offset", "io_callback", "size"]
+optional = ["offset", "io_callback", "size", "file_version"]
+
+[namespace.storage.op.stat]
+optional = ["file_version", "object_type"]
```

Comment:
ditto.

And we have `ObjectMode`, `object_type` is not needed.

Reply:
I couldn't found `ObjectMode` in stat ops, where should I found?
I separate these two types because different object type need different operations in minio-client, and stat to a folder object may cost more time.

Reply:
[GSP-93](https://github.com/beyondstorage/specs/blob/master/rfcs/93-add-object-mode-pair.md) has been approved, we can use ObjectMode in stat now.

minio doesn't have native support for `DirObject`, so we need to simulate it. Let's implement in another PR.

For current PR, It's OK to implement just like no dir object exists.
And in next PR, we need to:

- Implement CreateDir with `simulate = true` (described in [GSP-87](https://github.com/beyondstorage/specs/blob/master/rfcs/87-feature-gates.md))
- Implement Stat / Delete a dir object.

NOTICE: go-storage requires zero-cost abstraction. so we can't use ListObjects API in `stat`.